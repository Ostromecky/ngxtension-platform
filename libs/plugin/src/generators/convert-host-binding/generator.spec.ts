import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import { convertHostBindingGenerator } from './generator';
import { ConvertHostBindingGeneratorSchema } from './schema';

const filesMap = {
	componentWithHostBinding: `
	import { Component, HostBinding } from '@angular/core';

  @Component({
    selector: 'my-component',
    template: 'My component',
  })
  export class MyComponent {
    @HostBinding('class.active') isActive = true;

    @HostBinding('attr.aria-disabled') get isDisabled() {
      return true;
    }

    @HostBinding('tabIndex') getTabIndex() {
      return this.isDisabled ? -1 : 0;
    }
  }
  `,
	directiveWithHostBinding: `
	import { Directive, HostBinding } from '@angular/core';

  @Directive({
    selector: '[myDirective]'
  })
  export class MyDirective {
    @HostBinding('class.active') isActive = true;

    @HostListener('keydown', ['$event'])
    updateValue(event: KeyboardEvent) {
    }
  }
  `,

	componentWithHostProperty: `
  import { Component } from '@angular/core';

    @Component({
      selector: 'my-component',
      template: 'My component',
      host: {
        '[class.active]': 'isActive',
      }
    })
    export class MyComponent {
      isActive = true;

      @HostBinding('attr.aria-disabled') get isDisabled() {
        return true;
      }
    }
  `,
};

describe('convert-host-binding generator', () => {
	let tree: Tree;
	const options: ConvertHostBindingGeneratorSchema = {
		path: 'libs/my-file.ts',
	};

	function setup(file: keyof typeof filesMap) {
		tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
		tree.write('package.json', `{"dependencies": {"@angular/core": "17.3.0"}}`);
		tree.write(`libs/my-file.ts`, filesMap[file]);

		return () => {
			return [tree.read('libs/my-file.ts', 'utf8'), filesMap[file]];
		};
	}

	beforeEach(() => {
		tree = createTreeWithEmptyWorkspace();
	});

	it('should convert properly for component', async () => {
		const readContent = setup('componentWithHostBinding');
		await convertHostBindingGenerator(tree, options);

		const [updated] = readContent();

		expect(updated).toMatchSnapshot();
	});

	it('should convert properly for directive', async () => {
		const readContent = setup('directiveWithHostBinding');
		await convertHostBindingGenerator(tree, options);

		const [updated] = readContent();

		expect(updated).toMatchSnapshot();
	});

	it('should convert properly for component with host property', async () => {
		const readContent = setup('componentWithHostProperty');
		await convertHostBindingGenerator(tree, options);

		const [updated] = readContent();

		expect(updated).toMatchSnapshot();
	});
});
