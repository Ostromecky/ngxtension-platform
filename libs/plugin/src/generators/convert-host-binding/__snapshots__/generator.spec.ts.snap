// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert-host-binding generator should convert properly for component 1`] = `
"
	import { Component } from '@angular/core';

  @Component({
    selector: 'my-component',
    template: 'My component',
      host: { '[class.active]': 'isActive', '[attr.aria-disabled]': 'isDisabled', '[tabIndex]': 'getTabIndex()' }
})
  export class MyComponent {
    isActive = true;

    get isDisabled() {
      return true;
    }

    getTabIndex() {
      return this.isDisabled ? -1 : 0;
    }
  }
  "
`;

exports[`convert-host-binding generator should convert properly for component with host property 1`] = `
"
  import { Component } from '@angular/core';

    @Component({
      selector: 'my-component',
      template: 'My component',
      host: {
        '[class.active]': 'isActive',
          '[attr.aria-disabled]': 'isDisabled'
    }
    })
    export class MyComponent {
      isActive = true;

      get isDisabled() {
        return true;
      }
    }
  "
`;

exports[`convert-host-binding generator should convert properly for directive 1`] = `
"
	import { Directive } from '@angular/core';

  @Directive({
    selector: '[myDirective]',
      host: { '[class.active]': 'isActive',
          '(keydown)': 'updateValue($event)'
    }
})
  export class MyDirective {
    isActive = true;

    
    updateValue(event: KeyboardEvent) {
    }
  }
  "
`;
